src/System.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/TvpTest.cs

```diff
@@ -42,12 +43,36 @@ public class TvpTest
         private string _connStr;
 
         [ActiveIssue(27858, TestPlatforms.AnyUnix)]
-        [ConditionalFact(typeof(DataTestUtility),nameof(DataTestUtility.AreConnStringsSetup))]
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
         public void TestMain()
         {
             Assert.True(RunTestCoreAndCompareWithBaseline());
         }
 
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void TestPacketNumberWraparound()
+        {
+            // this test uses a specifically crafted sql record enumerator and data to put the TdsParserStateObject.WritePacket(byte,bool)
+            // into a state where it can't differentiate between a packet in the middle of a large packet-set after a byte counter wraparound
+            // and the first packet of the connection and in doing so trips over a check for packet length from the input which has been 
+            // forced to tell it that there is no output buffer space left, this causes an uncancellable infinite loop
+
+            // if the enumerator is completely read to the end then the bug is no longer present and the packet creation task returns,
+            // if the timeout occurs it is probable (but not absolute) that the write is stuck
+
+            var enumerator = new WraparoundRowEnumerator(1000000);
+
+            int returned = Task.WaitAny(
+                Task.Factory.StartNew(
+                    () => RunPacketNumberWraparound(enumerator),
+                    TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning
+                ),
+                Task.Delay(TimeSpan.FromSeconds(5))
+            );
+            Assert.True(enumerator.MaxCount == enumerator.Count);
```

Comment:
Re: Failure on Linux,
It may help to output the enumerator.Count on the console when the failure happens so that we know where really is this failing.

Reply:
I've upped the time from 5 seconds to 1 minute and added console debug info of state and time on failures. Lets see how that works.

Reply:
@Wraith2 The increase in delay worked. CI passes now.